src/github/logic.py

```diff
@@ -69,23 +80,53 @@ def _pull_request_commenters(pull_request: PullRequest) -> List[str]:
     return sorted(comment.author_handle() for comment in pull_request.comments())
 
 
-def pull_request_approved_before_merging(pull_request: PullRequest) -> bool:
+def pull_request_approved_before_merging(
+    pull_request: PullRequest,
+) -> ApprovedBeforeMergeStatus:
     """
     The pull request has been approved if the last review (approval/changes
-    requested) before merging was an approval
+    requested) before merging was an approval, ignoring reviews from users that
+    are marked as needing follow-up review.
     """
+    assert pull_request.merged(), "Checked for pre-merge approval on a non-merged PR"
     merged_at = pull_request.merged_at()
-    if merged_at is not None:
-        premerge_reviews = [
+    if merged_at is None:
```

Comment:
Is this a state we could actually get in, especially with the assertion added above?

Reply:
Unclear - there are separate `merged` and `merged_at` properties, so it seems possible but unlikely that we could hit this.